src/json-decoder.ts

```diff
@@ -251,6 +251,27 @@ export namespace JsonDecoder {
     });
   }
 
+  /**
+   * Tries to decode with `decoder` and returns `error` on failure, but allows
+   * for `undefined` or `null` values to be present at the top level and returns
+   * an `undefined` if the value was `undefined` or `null`.
+   *
+   * @param decoder The actual decoder to use.
+   */
+  export function optional<a>(
+    decoder: Decoder<a>
+  ): Decoder<a|undefined> {
+    return new Decoder<a|undefined>((json: any) => {
+      if (json === undefined) {
+        return ok<undefined>(undefined);
+      } else if (json === null) {
+        return ok<undefined>(undefined);
```

Comment:
Is there any reason why we'd prefer returning `null` instead of `undefined` here?